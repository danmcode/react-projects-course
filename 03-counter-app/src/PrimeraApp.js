import React from 'react';
import PropTypes from 'prop-types' //Obigar parametros


export const PrimeraApp = ({ saludo, subtitulo }) => {

    //const saludar = "Hola Mundo";

    return (
        <>
            <h1> { saludo } </h1>
            {/* <h1>  <pre> {JSON.stringify( { nombre: 'Daniel'}, null, 3 )} </pre> </h1> */}
            <p> {subtitulo} </p>
        </>
    );
};

/**
 * Definir el tipo de parametros que se necesitan
 * Asi como definir si el properties es requerido o no
 */

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
}

PrimeraApp.defaultProps = {
    subtitulo: "Subtitulo por defecto "
}

export default PrimeraApp;