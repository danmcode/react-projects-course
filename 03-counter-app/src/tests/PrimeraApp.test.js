import { shallow } from 'enzyme';
import React from "react";
import PrimeraApp from "../PrimeraApp";

describe('Pruebas en <PrimeraApp/>', () => { 
    
    // test('Debe mostrar el mensaje, Hola Soy Goku', () => { 
    //     const saludo = "Hola Soy Goku";

    //     const { getByText } = render(<PrimeraApp saludo={ saludo }/>);

    //     expect( getByText(saludo) ).toBeInTheDocument();
    //  });

    test('Debe mostrar el mensaje, Hola Soy Goku', () => {
        const saludo = "Hola, Soy Goku";
        const wrapper = shallow( <PrimeraApp saludo={ saludo }/> );

        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe mostrar un subtitulo enviado por Props', () => {
        const saludo = "Hola, Soy Goku";
        const subtitulo = "Soy un subtitulo";
        const wrapper = shallow( 
            <PrimeraApp 
                saludo={ saludo }
                subtitulo={ subtitulo }
            />
        );

        const textParrafo = wrapper.find('p').text();
        expect( textParrafo ).toBe(subtitulo);
        
    });
 });