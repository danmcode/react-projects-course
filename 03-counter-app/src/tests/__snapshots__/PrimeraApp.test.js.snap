// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp/> Debe mostrar el mensaje, Hola Soy Goku 1`] = `
<Fragment>
  <h1>
     
    Hola, Soy Goku
     
  </h1>
  <p>
    Subtitulo por defecto 
  </p>
</Fragment>
`;
